### DISCUSSION TITLE
Dynamic instrument switching

### DISCUSSION THREAD

**Discussion by 3hhh** [corrados/edrumulus#152] [Show and tell]
I recently realized that pretty much the only thing edrumulus and drumgizmo are worse at than commercial hardware is instrument switching.

So I wrote a [small xtalk plugin](https://github.com/3hhh/xtalk/blob/master/plugins/replace.py) which can be used to do exactly that by just replacing MIDI notes with others in real-time:
This way you can pre-load and map a virtual instrument/sound to an unused MIDI note in drumgizmo and then trigger the replacement of a previously used MIDI note to the pre-mapped instrument/sound. Triggering may happen via another MIDI note or a small TCP server as external API. The latter enables the use of e.g. USB foot switches.

Happy drumming!

**Comment by corrados** [comment]
I was also thinking about this a while ago. Thanks for providing the plugin. Only drawback is (I guess) that the initial loading of the system takes longer and you need a lot of memory. Thanks.

**Reply by 3hhh** [reply]
> Only drawback is (I guess) that the initial loading of the system takes longer and you need a lot of memory.

Yes, exactly.

> Thanks.

You're very welcome.

Btw I also wrote a [command-line metronome](https://github.com/3hhh/klickly) a while back. I thought I had mentioned it, but apparently I forgot.
